op-acceptor/runner/runner.go

```diff
@@ -504,52 +600,194 @@ func (r *runner) runSingleTest(metadata types.ValidatorMetadata) (*types.TestRes
 		"timeout", r.timeout,
 		"allowSkips", r.allowSkips)
 
+	// Initialize result
 	result := &types.TestResult{
 		Metadata: metadata,
 		Status:   types.TestStatusPass, // Default to pass unless determined otherwise
 	}
 
-	err := cmd.Run()
+	return result, cmd, testID, testBuffers{&stdout, &stderr}
+}
 
-	// Handle different error cases
+// logTestHeader logs the test header information
+func (r *runner) logTestHeader(testID, packageName, command string, startTime time.Time) {
+	// Log test header with structured logger
+	r.log.Debug("Starting test",
+		"test", testID,
+		"package", packageName,
+		"command", command,
+		"startTime", startTime)
+}
+
+// logTestOutput logs the stdout and stderr output
+func (r *runner) logTestOutput(stdout, stderr string) {
+	// Log detailed outputs at trace level
+	if stdout != "" {
+		r.log.Debug("Test stdout", "output", stdout)
+	}
+	if stderr != "" {
+		r.log.Debug("Test stderr", "output", stderr)
+	}
+}
+
+// processTestResult processes the result of a test execution
+func (r *runner) processTestResult(result *types.TestResult, ctx context.Context, err error, stdout, stderr string) (*types.TestResult, error) {
+	// Determine test status first based on context and error conditions
 	switch {
 	case ctx.Err() == context.DeadlineExceeded:
 		// Test timed out
 		result.Status = types.TestStatusFail
 		result.Error = fmt.Errorf("test timed out after %v", r.timeout)
+		r.log.Warn("Test timed out", "timeout", r.timeout)
 
 	case err != nil:
-		// Command failed but may just indicate test failure (not a Go execution error)
-		r.log.Debug("Test failed or had error",
-			"test", metadata.FuncName,
-			"package", metadata.Package,
-			"error", err)
-
+		// Command failed but may indicate test failure (not a Go execution error)
 		if exitErr, ok := err.(*exec.ExitError); ok {
 			result.Status = types.TestStatusFail
-			result.Error = fmt.Errorf("%s\n%s", exitErr.Error(), stderr.String())
+
+			// Look for more specific errors in stdout/stderr
+			detailedError := extractDetailedError(stdout, stderr)
+			if detailedError != "" {
+				result.Error = fmt.Errorf("%s\n%s", exitErr.Error(), detailedError)
+			} else {
+				result.Error = fmt.Errorf("%s\n%s", exitErr.Error(), stderr)
+			}
+			r.log.Debug("Test failed with exit error", "error", exitErr.Error(), "details", detailedError)
 		} else {
 			// Actual system error running the test command
-			return nil, fmt.Errorf("failed to execute test %s: %w", metadata.FuncName, err)
+			r.log.Error("System error running test", "error", err)
+			return nil, fmt.Errorf("failed to execute test %s: %w", result.Metadata.FuncName, err)
 		}
 
 	default:
 		// Command succeeded - check output for SKIP marker
-		output := stdout.String()
-		if strings.Contains(output, "--- SKIP:") {
+		if strings.Contains(stdout, "--- SKIP:") {
```

Comment:
I think we're fast approaching the "time to move to json output" threshold :)

Reply:
Oh yes, I think we're past that threshold! :)
Hope to send a PR for that change this week.